tests/Doctrine/Tests/Common/Util/ClassUtilsTest.php

```diff
@@ -65,6 +65,26 @@ public function testNewReflectionObject($className, $expectedClassName)
             $reflClass = ClassUtils::newReflectionObject($object);
             self::assertEquals($expectedClassName, $reflClass->getName());
         }
+
+        /**
+         * @runInSeparateProcess
+         * @preserveGlobalState disabled
+         */
+        public function testExistingClassNameWithColonIsNotDetectedAsAlias()
+        {
+            class_alias(self::class, 'a:b');
```

Comment:
This is a really bad idea - I don't expect the doctrine components to support this stuff explicitly :-\

Reply:
I may not follow you. This is here just to avoid PHP's engine dependent anonymous classes FQN's (which sometimes give `:` and sometimes don't) and to be able to test that case

Reply:
Don't think we should support anything around `get_class()` with anonymous classes. If you want to go down that path,  reflecting + checking whether a class is anonymous would be better.

Reply:
I agree, but this issue came out with Symfony validators that can validate arbitrary data, one of which may be anonymous classes. 

So it somehow made sense to report it here because right now, heuristic for aliased classes detects certain anonymous classes as aliased and because of that, exception is thrown.

To sum this up, it makes much more sense to go with the `::class` than to go with this heuristics?